src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.targets

```
@@ -388,9 +388,11 @@ The .NET Foundation licenses this file to you under the MIT license.
 
   <Target Name="CheckWasmSdks">
     <Error Text="Emscripten not found, not compiling to WebAssembly. To enable WebAssembly compilation, install Emscripten and ensure the EMSDK environment variable points to the directory containing upstream/emscripten/emcc.bat"
-           Condition="'$(EMSDK)' == ''" />
+           Condition="'$(EMSDK)' == '' and '$(_targetOS)' == 'browser'" />
     <Error Text="Wasi SDK not found, not compiling to WebAssembly. To enable WebAssembly compilation, install Wasi SDK and ensure the WASI_SDK_PATH environment variable points to the directory containing share/wasi-sysroot"
            Condition="'$(WASI_SDK_PATH)' == '' and '$(_targetOS)' == 'wasi'" />
+    <Warning Text="The WASI SDK version is too low. Please use WASI SDK 22 or newer with a 64 bit Clang."
+             Condition="!Exists('$(WASI_SDK_PATH)/VERSION')" />
```

```suggestion
             Condition="!Exists('$(WASI_SDK_PATH)/VERSION') and '$(_targetOS)' == 'browser'" />
```
You probably worked this one out already.